geospatial/dted.ksy

```diff
@@ -0,0 +1,399 @@
+        size: 4
+        encoding: ASCII
+      - id: reserved_for_nima_use_only1
+        type: str
+        size: 1
+        encoding: ASCII
+      - id: reserved_for_future_use2
+        type: str
+        size: 31
+        encoding: ASCII
+      - id: multiple_accuracy_outline_flag
+        type: str
+        size: 2
+        encoding: ASCII
+      - id: subregion
+        type: acc_subregion
+        repeat: expr
+        repeat-expr: 9
+      - id: reserved_for_nima_use_only2
+        type: str
+        size: 18
+        encoding: ASCII
+      - id: reserved_for_future_use3
+        type: str
+        size: 69
+        encoding: ASCII
+  acc_subregion:
+    seq:
+      - id: absolute_horizontal_accuracy_or_subregion_in_meters
+        type: str
+        size: 4
+        encoding: ASCII
+      - id: absolute_vertical_accuracy_or_subregion_in_meters
+        type: str
+        size: 4
+        encoding: ASCII
+      - id: relative_horizontal_accuracy_of_subregion_in_meters
+        type: str
+        size: 4
+        encoding: ASCII
+      - id: relative_vertical_accuracy_of_subregion_in_meters
+        type: str
+        size: 4
+        encoding: ASCII
+      - id: number_of_coordinates_in_accuracy_subregion_outline
+        type: str
+        size: 2
+        encoding: ASCII
+      - id: coordinate_pair
+        type: coordinate_pair_ddmmss_sh
+        repeat: expr
+        repeat-expr: 14
+  data_record:
+    seq:
+      - id: recognition_sentinal
+        type: u1
+      - id: data_block_count
+        type: b24
+      - id: logitude_count
+        type: u2
+      - id: latitude_count
+        type: u2
+      - id: elevations
+        type: data_record_elevation
+        repeat: expr
+        repeat-expr: _parent.uhl.number_of_latitude_points.value
+      - id: checksum
+        type: s4
+  data_record_elevation:
+    seq:
+      - id: is_negative
+        type: b1
+      - id: value
+        type: b15
+    instances:
+      meters:
+        value: value * (is_negative? -1 : 1)
+  ascii_int_ssss:
+    seq:
+      - id: text
+        type: str
+        size: 4
+        encoding: ASCII
+    instances:
+      value:
+        value: text.to_i
+  coordinate_pair_ddmmss_sh:
+    seq:
+      - id: latitude
+        type: angle_ddmmss_sh
+      - id: longitude
+        type: angle_dddmmss_sh
+  coordinate_pair_ddmmssh:
+    seq:
+      - id: latitude
+        type: angle_ddmmssh
+      - id: longitude
+        type: angle_dddmmssh
+  angle_ddmmss_sh:
+    seq:
+      - id: degrees
+        type: str
+        size: 2
+        encoding: ASCII
+      - id: minutes
+        type: str
+        size: 2
+        encoding: ASCII
+      - id: seconds
+        type: str
+        size: 4
+        encoding: ASCII
+      - id: hemisphere
+        type: str
+        size: 1
+        encoding: ASCII
+    instances:
+      degree_decimal:
+        value: (degrees.to_i + minutes.to_i/60.0 + seconds.to_i/3600.0) * (hemisphere == 'N'? 1 : -1)
+  angle_dddmmss_sh:
```

Comment:
What does `_sh` mean?

Reply:
The added docs explain what it is, but it represents the hemisphere of the degrees minutes seconds representation.

Reply:
The actual format of the field this type represents is DDDMMSS.SH where:
* DDD is 000 - 180 degrees.
* MM is 00 - 59 minute.
* SS is 00 - 59 seconds.
* H is the hemisphere, N,S,E or W.

In that case is it correct of me to include this formatting into the type name?

Reply:
>In that case is it correct of me to include this formatting into the type name?

It is, but better to avoid it, if there is no ambiguity without it.

Reply:
Personally I think it is fairly clear with the docs what the type actually represents. An alternative would be something like angle_tenth_of_seconds, and angle_even_seconds.

Reply:
What do you think?

Reply:
IMHO: for `angle` fields it may make sense to leave as it is. The remark wqs about other fields having the same pattern, like `date_yymm`.

Reply:
The odd part is that each of the following types
```
angle_ddmmss_sh
angle_dddmmss_sh
angle_ddmmssh
angle_dddmmssh
```
is that they really represent the same thing, an angle.  The only difference is how the data is parsed. If I had exactly 1 way to parse that field then I would shorten it down to just "angle".  Or if I could have the same type parsed in 4 different ways then I would do that.  Actually looking at the docs, maybe this would be a good usage of a Parameteric type.  Perhaps something like this:
```
  angle:
    params:
      - id: len_degrees
        type: u1
      - id: has_decimal_point
        type: u1
    seq:
      - id: degrees
        type: str
        size: len_degrees
      - id: minutes
        type: str
        size: 2
      - id: seconds
        type: str
        size: 2
      - id: decimal_point
        type: str
        size: 1
        if: has_decimal_point == 1
      - id: tenths_of_seconds
        type: str
        size: 1
        if: has_decimal_point == 1
      - id: hemisphere
        type: str
        size: 1
    instances:
      value:
        value: (degrees.to_i + minutes.to_i/60.0 + seconds.to_i/3600.0 + tenths_of_seconds.to_i/36000.0) * (hemisphere == 'N'? 1 : -1)
        doc: |
          Degree decimal floating point representation
```

Reply:
Unfortunately this means that I need to upgrade to 0.8 and the online repl won't work this this type anymore.